core/cap-0070.md

```diff
@@ -0,0 +1,141 @@
+## Preamble
+
+```
+CAP: 0070
+Title: Configurable SCP Timing Parameters
+Working Group:
+    Owner: Garand Tyson <@SirTyson>
+    Authors: Garand Tyson <@SirTyson>
+    Consulted: Nicolas Barry <@MonsieurNicolas>
+Status: Awaiting Decision
+Created: 2025-04-28
+Discussion: https://github.com/orgs/stellar/discussions/1719
+Protocol version: TBD
+```
+
+## Simple Summary
+
+This CAP introduces ledger configuration settings allowing the Stellar network to dynamically adjust ledger close times, nomination timeouts,
+and ballot timeouts to facilitate controlled, incremental improvements to block time performance.
+
+## Motivation
+
+Currently, Stellar's ledger close time and SCP timeout settings are hardcoded. This restricts the network's ability to gradually test and implement
+shorter block times, limiting improvements in performance and responsiveness. Additionally, as other network parameters such as ledger limits and transaction
+throughput evolve, SCP nomination and ballot phase timings may need to be adjusted to maintain efficiency and resilience.
+
+This CAP addresses these issues by making these parameters configurable, enabling small, incremental adjustments and performance tuning.
+
+## Goals Alignment
+
+The proposal aligns with Stellar’s goals of scalability, resilience, and performance by enabling iterative improvements in ledger close time and consensus efficiency.
+
+## Abstract
+
+We define new network configuration settings for:
+
+- Target ledger close time (milliseconds)
+- Initial nomination timeout (milliseconds)
+- Initial ballot timeout (milliseconds)
+- Per-round nomination timeout increment (milliseconds)
+- Per-round ballot timeout increment (milliseconds)
+
+These settings are introduced as ledger config parameters, allowing the Stellar network to adjust consensus timings incrementally.
+
+## Specification
+
+### XDR Changes
+
+```diff
```

Comment:
What is this diff based on? I tried to apply it to the curr and next branches of stellar-xdr with no success.

Reply:
This is based on next, `8903b65`. I removed some of the formatting from the actual diff, I'll add it back. I think that's probably what's giving you issues.

Reply:
FWIW you have to use patch for this, as CI doesn't seem to pass otherwise.

Reply:
I think CI has in issue. I tried to give it the stellar-xdr next branch, but it failed because it couldn't find the ref. It looks like the tool only looks at stellar-core and expects a stellar-core commit. I don't think this makes much sense, since core often lags behind XDR (and does so currently). I had to add `mddiffcheck.ignore=true` to pass the test.

Reply:
Ahh I think you have to pass the repo manually, default base is stellar-core. Let me give it another try

Reply:
Hmm so the tool supports specifying a repo, but the GI actions has stellar-core repo hardcoded. I've modified the action with a slightly hacky script that runs the check against both core (to preserve preexisting hashes) and stellar-xdr and fails only if both fail